staging/src/k8s.io/component-base/logs/datapol/externaltypes.go

```diff
@@ -42,6 +47,12 @@ func byType(t reflect.Type) []string {
 		return []string{"token"}
 	case x509Certificate:
 		return []string{"security-key"}
+	case oauth2Token, vendorPrefix + oauth2Token:
```

Comment:
why is vendor prefix added to these but not the ones above?

Reply:
is this method invoked for every log of every object? the Sprintf call is going to be very allocation heavy

referring to stringified versions of the types is vulnerable to drift, and we're already depending on them for the tests... why do we not want a `map[reflect.Type][]string`?

Reply:
1. Vendor prefix is used to handle difference in import path depending on build method (go and bazel). The types are golang builtin, so their import path is always the same.
2. This code is part of runtime log sanitization, it will analyse every logged object. Runtime sanitization is based on reflections that will be even more costly then this alloc. This mechanism is opt in and as of alpha is not meant for production use. 
3. As I expect that list of times will grow over time I wanted to avoid introducing those dependencies into the binary itself. Idea for using those dependencies in tests was that golang puts test dependencies in go.mod, but doesn't fetch them if not needed (same for bazel).

Reply:
if we're going to stick with string comparisons, avoid allocations in computing the key (which will be a miss for most inputs), and possibly even allocations in the return values, by doing something like this:

```
var mappings = map[string]map[string][]string{
  "net/http": map[string][]string{
    "Header": []string{"password", "token"},
    "Cookie": []string{"token"},
  },
  ...
}

func byType(t reflect.Type) []string {
  return mappings[t.PkgPath()][t.Name()]
}
```

Reply:
sg, want me to fix it in this PR?

Reply:
no, it's fine for follow-up

Reply:
> 3\. Idea for using those dependencies in tests was that golang puts test dependencies in go.mod, but doesn't fetch them if not needed (same for bazel).

It still passes those dependencies on to all downstream consumers regardless of whether they're vendored or not. See https://github.com/kubernetes/kubernetes/pull/96425#issuecomment-726464851 for the kind of pain that can cause. Let's not include them as test dependencies either